examples/kubernetes-client-examples/src/main/java/org/apache/flink/examples/Basic.java

```diff
@@ -0,0 +1,80 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.examples;
+
+import org.apache.flink.kubernetes.operator.crd.FlinkDeployment;
+import org.apache.flink.kubernetes.operator.crd.spec.FlinkDeploymentSpec;
+import org.apache.flink.kubernetes.operator.crd.spec.FlinkVersion;
+import org.apache.flink.kubernetes.operator.crd.spec.JobManagerSpec;
+import org.apache.flink.kubernetes.operator.crd.spec.JobSpec;
+import org.apache.flink.kubernetes.operator.crd.spec.Resource;
+import org.apache.flink.kubernetes.operator.crd.spec.TaskManagerSpec;
+import org.apache.flink.kubernetes.operator.crd.spec.UpgradeMode;
+
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Map.entry;
+
+/** client code for ../basic.yaml. */
+public class Basic {
+    public static void main(String[] args) {
+        FlinkDeployment flinkDeployment = new FlinkDeployment();
+        flinkDeployment.setApiVersion("flink.apache.org/v1beta1");
+        flinkDeployment.setKind("FlinkDeployment");
+        ObjectMeta objectMeta = new ObjectMeta();
+        objectMeta.setName("advanced-ingress");
+        flinkDeployment.setMetadata(objectMeta);
+        FlinkDeploymentSpec flinkDeploymentSpec = new FlinkDeploymentSpec();
+        flinkDeploymentSpec.setFlinkVersion(FlinkVersion.v1_15);
+        flinkDeploymentSpec.setImage("flink:1.15");
+        Map<String, String> flinkConfiguration =
+                Map.ofEntries(entry("taskmanager.numberOfTaskSlots", "2"));
+        flinkDeploymentSpec.setFlinkConfiguration(flinkConfiguration);
+        flinkDeployment.setSpec(flinkDeploymentSpec);
+        flinkDeploymentSpec.setServiceAccount("flink");
+        JobManagerSpec jobManagerSpec = new JobManagerSpec();
+        jobManagerSpec.setResource(new Resource(1.0, "2048m"));
+        flinkDeploymentSpec.setJobManager(jobManagerSpec);
+        TaskManagerSpec taskManagerSpec = new TaskManagerSpec();
+        taskManagerSpec.setResource(new Resource(1.0, "2048m"));
+        flinkDeploymentSpec.setTaskManager(taskManagerSpec);
+        flinkDeployment
+                .getSpec()
+                .setJob(
+                        JobSpec.builder()
+                                .jarURI(
+                                        "local:///opt/flink/examples/streaming/StateMachineExample.jar")
+                                .parallelism(2)
+                                .upgradeMode(UpgradeMode.STATELESS)
+                                .build());
+
+        try (KubernetesClient kubernetesClient =
```

Comment:
The example should work as is with minikube, using: KubernetesClient kubernetesClient = new DefaultKubernetesClient(); 

It is fine to add a utility explaining how to connect to a remote Kube API server, but please explain, how to get the token, cert, etc.